src/main/java/spireMapOverhaul/zones/gremlincamp/GremlinCamp.java

```diff
@@ -0,0 +1,202 @@
+package spireMapOverhaul.zones.gremlincamp;
+
+import basemod.ReflectionHacks;
+import com.badlogic.gdx.graphics.Color;
+import com.megacrit.cardcrawl.actions.common.ApplyPowerAction;
+import com.megacrit.cardcrawl.actions.unique.IncreaseMaxHpAction;
+import com.megacrit.cardcrawl.core.CardCrawlGame;
+import com.megacrit.cardcrawl.dungeons.AbstractDungeon;
+import com.megacrit.cardcrawl.events.shrines.GremlinMatchGame;
+import com.megacrit.cardcrawl.events.shrines.GremlinWheelGame;
+import com.megacrit.cardcrawl.helpers.MonsterHelper;
+import com.megacrit.cardcrawl.localization.UIStrings;
+import com.megacrit.cardcrawl.map.MapRoomNode;
+import com.megacrit.cardcrawl.monsters.AbstractMonster;
+import com.megacrit.cardcrawl.monsters.MonsterGroup;
+import com.megacrit.cardcrawl.monsters.exordium.*;
+import com.megacrit.cardcrawl.powers.AngerPower;
+import com.megacrit.cardcrawl.powers.BufferPower;
+import com.megacrit.cardcrawl.powers.RegenerateMonsterPower;
+import com.megacrit.cardcrawl.powers.RitualPower;
+import com.megacrit.cardcrawl.random.Random;
+import com.megacrit.cardcrawl.rooms.EventRoom;
+import com.megacrit.cardcrawl.rooms.ShopRoom;
+import com.megacrit.cardcrawl.ui.campfire.AbstractCampfireOption;
+import com.megacrit.cardcrawl.ui.campfire.RestOption;
+import spireMapOverhaul.SpireAnniversary6Mod;
+import spireMapOverhaul.abstracts.AbstractZone;
+import spireMapOverhaul.util.ActUtil;
+import spireMapOverhaul.util.Wiz;
+import spireMapOverhaul.zoneInterfaces.CampfireModifyingZone;
+import spireMapOverhaul.zoneInterfaces.CombatModifyingZone;
+import spireMapOverhaul.zoneInterfaces.EncounterModifyingZone;
+import spireMapOverhaul.zoneInterfaces.ModifiedEventRateZone;
+import spireMapOverhaul.zones.gremlincamp.monsters.GremlinBodyguard;
+import spireMapOverhaul.zones.gremlincamp.monsters.GremlinCook;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import static spireMapOverhaul.SpireAnniversary6Mod.makeID;
+import static spireMapOverhaul.SpireAnniversary6Mod.zoneEvents;
+
+public class GremlinCamp extends AbstractZone implements EncounterModifyingZone, CombatModifyingZone, CampfireModifyingZone, ModifiedEventRateZone {
```

Comment:
How do you intend for this zone to play out in terms of balance? It looks like it's mostly a balance-neutral zone with various tradeoffs (? nodes instead of shop nodes and no resting, but the fights don't seem that bad). Not making changes to rewards also suggests that it should aim for being balance neutral.

Reply:
Yeah, I consider this a theme park zone that just throws some different stuff at you. Don't think that every zone needs to change the rewards meaningfully.